File: codes/202603/10/sentiment_analyzer.py
# sentiment_analyzer.py
from utils import preprocess_text

# 간단한 규칙 기반 감성 사전 (예시)
# 실제 산업 프로젝트에서는 더 복잡한 사전, 기계 학습 모델 또는 딥러닝 모델을 사용합니다.
positive_keywords = ['좋은', '최고', '행복', '만족', '훌륭', '사랑', '긍정적', '기쁘다', '성공']
negative_keywords = ['나쁜', '최악', '슬픔', '불만', '실망', '혐오', '부정적', '화나다', '실패']

def analyze_sentiment(text):
    """
    주어진 텍스트의 감성을 분석합니다.
    여기서는 간단한 키워드 매칭 방식을 사용합니다.
    """
    tokens = preprocess_text(text)
    
    positive_count = sum(1 for token in tokens if token in positive_keywords)
    negative_count = sum(1 for token in tokens if token in negative_keywords)
    
    if positive_count > negative_count:
        return "긍정적"
    elif negative_count > positive_count:
        return "부정적"
    else:
        return "중립"

File: codes/202605/24/sentimentAnalyzer.js
// sentimentAnalyzer.js
// 텍스트의 감성을 분석하는 모듈이다.
// 'natural' 라이브러리를 사용하려 했으나, 한국어 감성 분석을 위한 복잡한 설정 없이
// 실행 가능한 예시를 위해 간단한 키워드 기반 감성 분석을 구현한다.
// 실제 산업에서는 훈련된 AI/ML 모델(예: BERT, KoBERT 등)을 활용한다.

/**
 * 텍스트의 감성을 분석하여 '긍정', '부정', '중립' 중 하나를 반환한다.
 * 매우 기본적인 키워드 매칭 방식을 사용한다.
 * @param {string} text - 분석할 텍스트 문자열
 * @returns {'긍정' | '부정' | '중립'} - 분석된 감성
 */
function analyzeSentiment(text) {
  const positiveKeywords = ["훌륭합니다", "만족합니다", "좋습니다", "추천합니다", "최고예요", "추천", "개선"];
  const negativeKeywords = ["실망스럽습니다", "좋지 않습니다", "아까워요", "별로입니다", "문제", "나쁘지 않지만"];

  let positiveScore = 0;
  let negativeScore = 0;

  // 긍정 키워드 검색
  positiveKeywords.forEach(keyword => {
    if (text.includes(keyword)) {
      positiveScore++;
    }
  });

  // 부정 키워드 검색
  negativeKeywords.forEach(keyword => {
    if (text.includes(keyword)) {
      negativeScore++;
    }
  });

  if (positiveScore > negativeScore) {
    return '긍정';
  } else if (negativeScore > positiveScore) {
    return '부정';
  } else {
    // 점수가 같거나, 아무 키워드도 발견되지 않은 경우
    return '중립';
  }
}

module.exports = {
  analyzeSentiment
};

/*
// [참고] 'natural' 라이브러리를 사용한 영어 감성 분석 예시 (한국어는 별도 모델 필요)
const natural = require('natural');
const Analyzer = natural.SentimentAnalyzer;
const stemmer = natural.PorterStemmer;
const analyzer = new Analyzer("English", stemmer, "afinn");

function analyzeEnglishSentiment(text) {
  // 텍스트를 토큰화한 후 감성 점수 계산
  const tokens = new natural.WordTokenizer().tokenize(text);
  const sentimentScore = analyzer.getSentiment(tokens);

  if (sentimentScore > 0.5) {
    return '긍정';
  } else if (sentimentScore < -0.5) {
    return '부정';
  } else {
    return '중립';
  }
}
*/
